src/channels/message_sender.py

```
@@ -0,0 +1,301 @@
+            except ValueError:
+                if self.instance_config.discord_default_channel_id:
+                    channel_id = int(self.instance_config.discord_default_channel_id)
+                else:
+                    return {"success": False, "error": "Invalid Discord channel ID", "channel": "discord"}
+            
+            # Prepare attachment
+            attachments = []
+            if media_url:
+                # Download media from URL
+                async with aiohttp.ClientSession() as session:
+                    async with session.get(media_url) as response:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)